src/Sylius/Bundle/AdminBundle/Controller/ArchiveCatalogPromotionAction.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\AdminBundle\Controller;
+
+use Sylius\Bundle\CoreBundle\CatalogPromotion\Processor\CatalogPromotionArchivalProcessorInterface;
+use Sylius\Component\Promotion\Exception\CatalogPromotionAlreadyArchivedException;
+use Sylius\Component\Promotion\Exception\CatalogPromotionNotFoundException;
+use Sylius\Component\Promotion\Exception\InvalidCatalogPromotionStateException;
+use Symfony\Component\HttpFoundation\Exception\BadRequestException;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\Session\Session;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+final class ArchiveCatalogPromotionAction
+{
+    public function __construct(private CatalogPromotionArchivalProcessorInterface $catalogPromotionArchivalProcessor)
+    {
+    }
+
+    public function __invoke(Request $request): Response
+    {
+        /** @var Session $session */
```

Comment:
If you need this phpdoc there is something wrong with the deps somewhere

Reply:
It returns `Session|callable`, so PHPStan highlights is as an issue.